File: vandalizer-next/backend/app/models/document.py
import datetime
from typing import Optional

from pydantic import Field
from beanie import Document


class SmartDocument(Document):
    path: str
    downloadpath: str
    processing: bool = False
    validating: bool = False
    valid: bool = True
    validation_feedback: Optional[str] = None
    task_id: Optional[str] = None
    task_status: Optional[str] = None
    title: str
    raw_text: str = ""
    extension: str = "pdf"
    uuid: str
    space: str
    user_id: str
    created_at: datetime.datetime = Field(default_factory=datetime.datetime.now)
    updated_at: datetime.datetime = Field(default_factory=datetime.datetime.now)
    folder: Optional[str] = None
    is_default: bool = False
    token_count: int = 0
    num_pages: int = 0

    class Settings:
        name = "smart_document"

File: frontend/src/types/document.ts
export interface Document {
  id: string
  title: string
  uuid: string
  extension: string
  processing: boolean
  valid: boolean
  task_status: string | null
  folder: string | null
  created_at: string
  token_count: number
  num_pages: number
}

export interface Folder {
  id: string
  title: string
  uuid: string
  parent_id: string
  is_shared_team_root: boolean
  team_id?: string | null
}

export interface ListContentsResponse {
  folders: Folder[]
  documents: Document[]
}

export interface PollStatusResponse {
  status: string | null
  status_messages: string[]
  complete: boolean
  raw_text: string
  validation_feedback: string | null
  valid: boolean
  path: string | null
}
